### PR TITLE
Adding traces around User service close async and reducing ttl for slow operation completions

### PR DESCRIPTION
**Pull request by amanbha**
Fixes https://github.com/microsoft/service-fabric-services-and-actors-dotnet/issues/249

### CODE REVIEW

File: src/Microsoft.ServiceFabric.Services/Runtime/StatefulServiceReplicaAdapter.cs [hunk 1]
```diff
@@ -195,11 +195,31 @@ async Task IStatefulServiceReplica.CloseAsync(CancellationToken cancellationToke
 
             await this.CloseCommunicationListenersAsync(cancellationToken);
             await this.CancelRunAsync();
+            ServiceTrace.Source.WriteInfoWithId(
```
Comment (LMWF) [inline src/Microsoft.ServiceFabric.Services/Runtime/StatefulServiceReplicaAdapter.cs:198]: Would it be useful logging replica/partition id here too?

File: src/Microsoft.ServiceFabric.Services/Runtime/StatefulServiceReplicaAdapter.cs [hunk 2]
```diff
@@ -195,11 +195,31 @@ async Task IStatefulServiceReplica.CloseAsync(CancellationToken cancellationToke
...
             await this.userServiceReplica.OnCloseAsync(cancellationToken);
 
+            ServiceTrace.Source.WriteInfoWithId(
+                TraceType,
+                this.traceId,
+                "Completed call to userServiceReplica.OnCloseAsync().");
+
             if (this.stateProviderReplica != null)
             {
+                ServiceTrace.Source.WriteInfoWithId(
```
Comment (LMWF) [inline src/Microsoft.ServiceFabric.Services/Runtime/StatefulServiceReplicaAdapter.cs:212]: The indentation here and on 219 looks odd.

File: src/Microsoft.ServiceFabric.Services/Runtime/StatefulServiceReplicaAdapter.cs (hunk 1 above)
Comment (amanbha) [reply src/Microsoft.ServiceFabric.Services/Runtime/StatefulServiceReplicaAdapter.cs:198]: its already logged in the type field.

File: src/Microsoft.ServiceFabric.Services/Runtime/StatefulServiceReplicaAdapter.cs (hunk 2 above)
Comment (amanbha) [reply src/Microsoft.ServiceFabric.Services/Runtime/StatefulServiceReplicaAdapter.cs:212]: Fixed.

File: src/Microsoft.ServiceFabric.Services/Runtime/StatefulServiceReplicaAdapter.cs (hunk 1 above)
Comment (LMWF) [reply src/Microsoft.ServiceFabric.Services/Runtime/StatefulServiceReplicaAdapter.cs:198]: ok

**Pull request merged**